bug: Gateway/GatewayClass analyzers panic on objects with no status conditions

## Expected Behavior

A Gateway or GatewayClass that has no `.status.conditions` yet (a freshly created object before its controller writes status, or a cluster with no Gateway controller installed) should be skipped by the analyzer with no failure reported, just like the PodDisruptionBudget analyzer already does after #664.

## Current Behavior

`k8sgpt analyze` crashes the entire run with:

```
panic: runtime error: index out of range [0] with length 0
```

The Gateway analyzer reads `gtw.Status.Conditions[0]` and the GatewayClass analyzer reads `gc.Status.Conditions[0]` with no length check, so any Gateway/GatewayClass that has not been reconciled yet brings down the whole analyze run.

- `pkg/analyzer/gateway.go:77`
- `pkg/analyzer/gatewayclass.go:56`

## Steps to Reproduce

1. Apply a `Gateway` (or `GatewayClass`) resource to a cluster that has no Gateway controller installed, so its `.status.conditions` stays empty.
2. Run `k8sgpt analyze`.
3. The command panics with `index out of range [0] with length 0` instead of reporting/skipping the resource.

This is also confirmed via the analyzer unit tests: reverting the length-check guard makes the empty-conditions test panic with `index out of range [0] with length 0`.

## Environment

- **k8sgpt version:** main (`go build` from current `main`)
- **Kubernetes version:** any (reproduces against the Gateway API types directly)
- **AI Backend/Provider:** N/A (panic happens during analysis, before any AI call)
- **OS/Platform:** any; reproduced on macOS with Go 1.24+

## Additional Context

The PodDisruptionBudget analyzer hit the same class of bug and was fixed in #664 by checking `len(...Status.Conditions)` before indexing `[0]`. The Gateway and GatewayClass analyzers never received that guard. Fix incoming that mirrors the PDB precedent.

--- a/pkg/analyzer/gateway.go
+++ b/pkg/analyzer/gateway.go
@@ -74,7 +74,7 @@ func (GatewayAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
 
 		// Check only the current conditions
 		// TODO: maybe check other statuses Listeners, addresses?
-		if gtw.Status.Conditions[0].Status != metav1.ConditionTrue {
+		if len(gtw.Status.Conditions) > 0 && gtw.Status.Conditions[0].Status != metav1.ConditionTrue {
 			failures = append(failures, common.Failure{
 				Text: fmt.Sprintf("Gateway '%s/%s' is not accepted. Message: '%s'.",
 					gtwNamespace,

--- a/pkg/analyzer/gateway_test.go
+++ b/pkg/analyzer/gateway_test.go
@@ -91,6 +91,49 @@ func TestGatewayAnalyzer(t *testing.T) {
 
 }
 
+func TestEmptyConditionsGatewayAnalyzer(t *testing.T) {
+	ClassName := gtwapi.ObjectName("exists")
+	AcceptedStatus := metav1.ConditionTrue
+	GatewayClass := BuildGatewayClass(string(ClassName))
+
+	Gateway := BuildGateway(ClassName, AcceptedStatus, nil)
+	// A freshly created Gateway (or one whose controller has not reported yet)
+	// has no status conditions; the analyzer must not panic on it.
+	Gateway.Status.Conditions = nil
+	// Create a Gateway Analyzer instance with the fake client
+	scheme := scheme.Scheme
+
+	err := gtwapi.Install(scheme)
+	if err != nil {
+		t.Error(err)
+	}
+	err = apiextensionsv1.AddToScheme(scheme)
+	if err != nil {
+		t.Error(err)
+	}
+	objects := []runtime.Object{
+		&Gateway,
+		&GatewayClass,
+	}
+
+	fakeClient := fakeclient.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(objects...).Build()
+
+	analyzerInstance := GatewayAnalyzer{}
+	config := common.Analyzer{
+		Client: &kubernetes.Client{
+			CtrlClient: fakeClient,
+		},
+		Context:   context.Background(),
+		Namespace: "default",
+	}
+	analysisResults, err := analyzerInstance.Analyze(config)
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, len(analysisResults), 0)
+
+}
+
 func TestMissingClassGatewayAnalyzer(t *testing.T) {
 	ClassName := gtwapi.ObjectName("non-existed")
 	AcceptedStatus := metav1.ConditionTrue

--- a/pkg/analyzer/gatewayclass.go
+++ b/pkg/analyzer/gatewayclass.go
@@ -53,7 +53,7 @@ func (GatewayClassAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) 
 
 		gcName := gc.GetName()
 		// Check only the current condition
-		if gc.Status.Conditions[0].Status != metav1.ConditionTrue {
+		if len(gc.Status.Conditions) > 0 && gc.Status.Conditions[0].Status != metav1.ConditionTrue {
 			failures = append(failures, common.Failure{
 				Text: fmt.Sprintf(
 					"GatewayClass '%s' with a controller name '%s' is not accepted. Message: '%s'.",

--- a/pkg/analyzer/gatewayclass_test.go
+++ b/pkg/analyzer/gatewayclass_test.go
@@ -56,6 +56,41 @@ func TestGatewayClassAnalyzer(t *testing.T) {
 
 }
 
+// A GatewayClass with no status conditions (newly created, or no controller
+// installed) must not panic the analyzer.
+func TestEmptyConditionsGatewayClassAnalyzer(t *testing.T) {
+	GatewayClass := &gtwapi.GatewayClass{}
+	GatewayClass.Name = "foobar"
+	GatewayClass.Spec.ControllerName = "gateway.fooproxy.io/gatewayclass-controller"
+	// Create a GatewayClassAnalyzer instance with the fake client
+	scheme := scheme.Scheme
+	err := gtwapi.Install(scheme)
+	if err != nil {
+		t.Error(err)
+	}
+	err = apiextensionsv1.AddToScheme(scheme)
+	if err != nil {
+		t.Error(err)
+	}
+
+	fakeClient := fakeclient.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(GatewayClass).Build()
+
+	analyzerInstance := GatewayClassAnalyzer{}
+	config := common.Analyzer{
+		Client: &kubernetes.Client{
+			CtrlClient: fakeClient,
+		},
+		Context:   context.Background(),
+		Namespace: "default",
+	}
+	analysisResults, err := analyzerInstance.Analyze(config)
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, len(analysisResults), 0)
+
+}
+
 func TestGatewayClassAnalyzerLabelSelectorFiltering(t *testing.T) {
 	condition := metav1.Condition{
 		Type:    "Accepted",

